client_test.go

```diff
@@ -449,6 +450,103 @@ func TestSendMessageBatch(t *testing.T) {
 	assert.Nil(t, err)
 }
 
+func TestSendMessageBatchMessageSizesBelowThreshold(t *testing.T) {
+	ms3c := &mockS3Client{&mock.Mock{}}
+	msqsc := &mockSQSClient{Mock: &mock.Mock{}}
+	msqsc.On(
+		"SendMessageBatch",
+		mock.Anything,
+		mock.MatchedBy(func(params *sqs.SendMessageBatchInput) bool {
+			assert.Equal(t, "testing_url", *params.QueueUrl)
+			assert.Len(t, params.Entries, 2)
+			assert.Equal(t, "entry_1", *params.Entries[0].Id)
+			assert.Equal(t, "entry_2", *params.Entries[1].Id)
+			assert.Equal(t, "testing body 1", *params.Entries[0].MessageBody)
+			assert.Equal(t, "testing body 2", *params.Entries[1].MessageBody)
+			assert.Equal(t, "hi", *params.Entries[0].MessageAttributes["testing_attribute"].StringValue)
+			assert.Equal(t, "hello", *params.Entries[1].MessageAttributes["testing_attribute"].StringValue)
+			assert.Nil(t, params.Entries[0].MessageAttributes["ExtendedPayloadSize"].StringValue)
+			assert.Nil(t, params.Entries[1].MessageAttributes["ExtendedPayloadSize"].StringValue)
+			return true
+		}),
+		mock.Anything).
+		Return(&sqs.SendMessageBatchOutput{}, nil)
+
+	c, err := New(msqsc, ms3c, WithMessageSizeThreshold(500), WithS3BucketName("test_bucket"))
+	assert.Nil(t, err)
+
+	_, err = c.SendMessageBatch(context.Background(), &sqs.SendMessageBatchInput{
+		Entries: []types.SendMessageBatchRequestEntry{
+			{
+				Id:          aws.String("entry_1"),
+				MessageBody: aws.String("testing body 1"),
+				MessageAttributes: map[string]types.MessageAttributeValue{
+					"testing_attribute": {StringValue: aws.String("hi")},
+				},
+			},
+			{
+				Id:          aws.String("entry_2"),
+				MessageBody: aws.String("testing body 2"),
+				MessageAttributes: map[string]types.MessageAttributeValue{
+					"testing_attribute": {StringValue: aws.String("hello")},
+				},
+			},
+		},
+		QueueUrl: aws.String("testing_url|override_bucket"),
+	})
+
+	assert.Zero(t, ms3c.Calls)
+	assert.Len(t, msqsc.Calls, 1)
+	assert.Nil(t, err)
+}
+
+func TestSendMessageBatchSizeAboveThreshold(t *testing.T) {
```

Comment:
Tweaked this test case- first message is less than the BatchMessageSizeThreshold (`20`), but the second and third are larger than the configured threshold, so are sent as pointers.